src/backend/src/services/projects.services.ts

```diff
@@ -811,4 +811,38 @@ export default class ProjectsService {
 
     return newMaterialType;
   }
+
+  /**
+   * Assign a material on a project to a different assembly
+   * @param submitter the submitter
+   * @param materialId the material that will be moved
+   * @param assemblyId the assembly to change the material to
+   * @throws if the submitter does not have the relevant positions
+   */
+  static async assignMaterialAssembly(submitter: User, materialId: string, assemblyId: string) {
+    const material = await prisma.material.findUnique({ where: { materialId } });
+    if (!material) throw new NotFoundException('Material', materialId);
+
+    const assembly = await prisma.assembly.findUnique({
+      where: { assemblyId },
+      include: { wbsElement: true }
+    });
+    if (!assembly) throw new NotFoundException('Assembly', assemblyId);
+
+    const project = await prisma.project.findFirst({
+      where: {
+        wbsElement: assembly.wbsElement
+      },
+      ...projectQueryArgs
+    });
+    if (!project) throw new NotFoundException('Project', assembly.wbsElementId);
+
+    // Permission: leadership and up, anyone on project team
+    if (!(isLeadership(submitter.role) || isUserPartOfTeams(project?.teams, submitter)))
+      throw new AccessDeniedException('Only leadership or above can create a material type');
+
```

Comment:
I'm getting an issue with this in testing. Should I be mocking `isUserPartOfTeams` too?

Reply:
I don't think so. I think we just wanna make sure we mock the correct project and pass the correct submitter from the test for this.

Also, we can remove the question mark after the project on line 841 since we already verified it's defined from line 838.